sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_json.py

```diff
@@ -0,0 +1,94 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+ESCAPE_CHAR = "\\"
+DOUBLE_FORWARD_SLASH_COMMENT = "//"
+MULTI_LINE_COMMENT = "/*"
+END_MULTI_LINE_COMMENT = "*/"
+DOUBLE_QUOTE = '"'
+NEW_LINE = "\n"
+
+
+def _strip_ignore_string(text: str, i: int, result: list[str]) -> tuple[list[str], int]:
+    """
+    Helper function to strip comments within a string literal.
+    Handles both single and double quotes, and escaped quotes.
+    Returns the updated index and the result list.
+
+    :param text: The input text to process.
+    :type text: str
+    :param i: The current index in the text.
+    :type i: int
+    :param result: The list to accumulate the processed characters.
+    :type result: list[str]
+    :raises ValueError: If there is an unterminated string literal.
+    :return: A tuple containing the updated result list and the next index.
+    :rtype: tuple[list[str], int]
+    """
+    while i < len(text):
+        current_char = text[i]
+        result.append(current_char)
```

Comment:
It's a little odd to me that this function both mutates `result` that is passed in, and returns it as well. Returning it doesn't seem necessary given it's mutated.